lib/src/consts.rs

```diff
@@ -122,16 +125,54 @@ lazy_static! {
     };
 }
 
-pub fn get_network_spec(network: Network) -> ChainSpec {
-    match network {
-        Network::Ethereum => ETH_MAINNET_CHAIN_SPEC.clone(),
-        Network::Holesky => ETH_HOLESKY_CHAIN_SPEC.clone(),
-        Network::TaikoA7 => TAIKO_A7_CHAIN_SPEC.clone(),
+#[derive(Clone, Debug)]
+pub struct SupportedChainSpecs(HashMap<String, ChainSpec>);
+
+impl SupportedChainSpecs {
+    pub fn default() -> Self {
+        SupportedChainSpecs(
+            vec![
+                ETH_MAINNET_CHAIN_SPEC.clone(),
+                ETH_HOLESKY_CHAIN_SPEC.clone(),
+                TAIKO_A7_CHAIN_SPEC.clone(),
+            ]
+            .into_iter()
+            .map(|cs| (cs.name.clone(), cs))
+            .collect(),
+        )
+    }
+
+    #[cfg(feature = "std")]
+    pub fn merge_from_file(file_path: PathBuf) -> Self {
+        fn load_and_merge(file_path: PathBuf) -> Result<SupportedChainSpecs> {
+            let mut known_chain_specs = SupportedChainSpecs::default();
+            let file = std::fs::File::open(&file_path)?;
+            let reader = std::io::BufReader::new(file);
+            let config: Value = serde_json::from_reader(reader)?;
+            let chain_spec_list: Vec<ChainSpec> = serde_json::from_value(config)?;
+            let new_chain_specs = chain_spec_list
+                .iter()
+                .map(|cs| (cs.name.clone(), cs.clone()))
+                .collect::<HashMap<String, ChainSpec>>();
+
+            // override known specs
+            known_chain_specs.0.extend(new_chain_specs);
+            Ok(known_chain_specs)
+        }
+        load_and_merge(file_path).unwrap_or(SupportedChainSpecs::default())
+    }
```

Comment:
Just my way of thinking, but I would probably separate these two actions. When merging from a file, I expect that known specs won't be impacted if something fails, where here we return a default result even if we had a working set of specs before the file reading, which might work for the current use case, but this is now built into the `merge_from_file` function and we couldn't get different behavior.

Reply:
I see that, didn't have an issue with that, just that hopefully they wouldn't always get overriden if there was an error with the file read. This works great now!